src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -1,715 +1,29 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
-
-using System;
-using System.Collections.Concurrent;
+﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
-using System.Threading;
-using System.Threading.Tasks;
+using System.Web;
 using System.Web.Configuration;
-using NuGet.Server.Logging;
 
 namespace NuGet.Server.Infrastructure
 {
-    /// <summary>
-    /// ServerPackageRepository represents a folder of nupkgs on disk. All packages are cached during the first request in order
-    /// to correctly determine attributes such as IsAbsoluteLatestVersion. Adding, removing, or making changes to packages on disk 
-    /// will clear the cache.
-    /// </summary>
-    public class ServerPackageRepository
-        : PackageRepositoryBase, IServerPackageRepository, IPackageLookup, IDisposable
+	public class ServerPackageRepository : ServerPackageRepositoryCore
```

Comment:
Primry reason for refactoring was that the old implementation was effectively hardcoded to read settings from Web.config (no public constructor where caller could provide getSetting callback). I did it like this to ensure 100% backward source compatibility with existing clients. If that is not neccessary, the easiest and cleanest approach might be: 
- remove the ServerPackageRepository class from NuGet.Server.
- rename ServerPackageRepositoryCore in NuGet.Server back to ServerPackageRepository
- Move GetBooleanAppSetting from ServerPackageRepository in NuGet.Server to the dataservice (Packages.svc.cs) or to a new (or existing) utility class.
- Modify DefaultServiceResolver in NuGet.Server to send GetBooleanAppSetting as argument to constructor of ServerPackageRepository